src/components/schedule-inputs.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ChangeEvent } from 'react';
+import tzdata from 'tzdata';
+
+import { Autocomplete, TextField } from '@mui/material';
+
+import { useTranslator } from '../hooks';
+
+export type ScheduleInputsProps = {
+  idPrefix: string;
+  schedule?: string;
+  timezone?: string;
+  handleScheduleChange: (event: ChangeEvent) => void;
+  handleTimezoneChange: (newValue: string | null) => void;
+};
+
+export function ScheduleInputs(props: ScheduleInputsProps): JSX.Element | null {
+  const trans = useTranslator('jupyterlab');
+
+  const timezones = Object.keys(tzdata.zones).sort();
+
+  const timezoneLabel = trans.__('Time zone');
+
+  return (
+    <>
+      <TextField
+        label={trans.__('Cron expression')}
```

Comment:
Is there a regular expression we can use the validate the CRON expression?

Reply:
There are several, which are hideously long: https://www.codeproject.com/Tips/5299523/Regex-for-cron-expressions

We could apply some simple regexes, like checking for non-blank values or checking for four spaces between the expression's components, as an alternative.

Reply:
[cron-validate](https://www.npmjs.com/package/cron-validate) can do it all for me.